File: app/src/main/java/co/tpcreative/supersafe/common/presenter/BaseServiceView.java
package co.tpcreative.supersafe.common.presenter;
import android.content.Context;

import co.tpcreative.supersafe.model.EnumStatus;
public interface BaseServiceView <T> {
    void onError(String message, EnumStatus status);
    void onSuccessful(String message,EnumStatus status);
    Context getContext();
}

File: app/src/main/java/co/tpcreative/supersafe/common/presenter/BaseServiceView.kt
package co.tpcreative.supersafe.common.presenter
import android.content.Context
import co.tpcreative.supersafe.model.EnumStatus

interface BaseServiceView<T> {
    fun onError(message: String?, status: EnumStatus)
    fun onSuccessful(message: String?, status: EnumStatus)
    fun getContext(): Context?
}